README.md

```diff
@@ -215,18 +245,80 @@ R> jt
 Phylogenetic tree with 10 tips and 9 internal nodes.
 
 Tip labels:
-	t3, t5, t8, t1, t10, t9, ...
+	t1, t4, t3, t6, t9, t7, ...
 
 Rooted; includes branch lengths.
 
-R> all.equal(rt, jt) # check no damage in translations
-[1] TRUE
+R> if (all.equal(rt, jt)) "no damage in translation"
+[1] "no damage in translation"
 ```
 
 For the time being the code will only work with rooted trees
 with named tips and branch lengths. If there's [demand][issues-url]
 for other types of trees, I'll look into it.
 
+### Trait evolution
+
+As far as traits are concerned, these can be continuous pr
+discrete. First a continuous trait:
+
+```julia
+julia> using Phylo, Plots, DataFrames
+
+julia> tree = rand(Nonultrametric(100)) # Defaults to mean tree depth of 1.0
+LinkTree{OneRoot,String,LinkNode{OneRoot,String,Dict{String,Any},LinkBranch{OneRoot,String,Dict{String,Any},Float64}},LinkBranch{OneRoot,String,Dict{String,Any},Float64},Dict{String,Any}} with 100 tips, 199 nodes and 198 branches.
+Leaf names are tip 21, tip 81, tip 32, tip 12, tip 51, ... [94 omitted] ... and tip 93
+
+julia> rand(BrownianTrait(tree, "Trait"))  # Defaults to starting at 0.0, variance 1.0
+LinkTree{OneRoot,String,LinkNode{OneRoot,String,Dict{String,Any},LinkBranch{OneRoot,String,Dict{String,Any},Float64}},LinkBranch{OneRoot,String,Dict{String,Any},Float64},Dict{String,Any}} with 100 tips, 199 nodes and 198 branches.
+Leaf names are tip 21, tip 81, tip 32, tip 12, tip 51, ... [94 omitted] ... and tip 93
+
+julia> plot(tree, line_z = getnodedata.(tree, traversal(tree, postorder), "Trait"))
+
+julia> d = DataFrame(nodename=getnodename.(tree, traversal(tree, preorder)), trait=getnodedata.(tree, traversal(tree, preorder), "Trait"))
+199×2 DataFrame
+│ Row │ nodename │ trait     │
+│     │ String   │ Float64   │
+├─────┼──────────┼───────────┤
+│ 1   │ Node 199 │ 0.0       │
+│ 2   │ Node 198 │ -0.517593 │
+│ 3   │ Node 197 │ -0.314101 │
+│ 4   │ Node 172 │ -0.391149 │
+│ 5   │ Node 164 │ -0.797994 │
+⋮
+│ 195 │ tip 67   │ -0.21145  │
+│ 196 │ Node 125 │ 0.236189  │
+│ 197 │ tip 79   │ 0.218236  │
+│ 198 │ tip 40   │ 0.745802  │
+│ 199 │ tip 81   │ -0.408055 │
+```
+
+Then a discrete trait:
+```julia
+julia> @enum TemperatureTrait lowTempPref midTempPref highTempPref
+
+julia> rand(SymmetricDiscreteTrait(tree, TemperatureTrait, 0.4));
```

Comment:
this `rand` interface seems a bit weird to me - should it not be `rand!` given that you're modifying the tree? I'd expect `rand` to return a vector or Dict with the trait

Reply:
I did agonise a bit about that, and I decided `rand!` wasn't right because I'm not providing the `tree` object directly store the trait in. But you're obviously right that `rand` is wrong too, and I was just ignoring that. Maybe the right answer is to return a `Dict` from `rand` and for `rand!` to have a second argument of a tree?

Reply:
Yeah, or maybe even better `rand!(tree, SymmetricDiscretTrait(TemperatureTrait, 0.4))`?

Reply:
and yes to a Dict from `rand` :+1:

Reply:
I think for consistency with other `rand!` calls, it would have to be `rand!(SymmetricDiscreteTrait(TemperatureTrait, 0.4), tree)`. However, dropping the `tree` parameter from `SymmetricDiscreteTrait` would mean that you couldn't call `rand(SymmetricDiscreteTrait(TemperatureTrait, 0.4))`... maybe it's okay if we just have `rand!` though? If not, I think it would have to be `rand!(SymmetricDiscreteTrait(tree, TemperatureTrait, 0.4), tree)` though...

Reply:
Why would you need to repeat the `tree` twice? Good point on the order

Reply:
You need `tree` twice if you want a `rand` function, surely? Because `rand(SymmetricDiscreteTrait(TemperatureTrait, 0.4))` doesn't work as there's no tree to evolve the trait on, so it has to be `rand(SymmetricDiscreteTrait(tree, TemperatureTrait, 0.4))`.

Reply:
First, of all, my name isn't Shirley. Second - yeah, good point :-p

Reply:
Sorry, couldn't help making my old "naked gun" joke on "surely" ...

Reply:
Nothing wrong with Naked Gun jokes :)